proc/threads.go

```diff
@@ -280,6 +280,12 @@ func (thread *Thread) GetG() (g *G, err error) {
 		return nil, err
 	}
 
+	if thread.dbp.arch.GStructOffset() == 0 {
+		// GetG was called through SwitchThread / updateThreadList during initialization
+		// thread.dbp.arch isn't setup yet (it needs a CurrentThread to read global variables from)
+		return nil, nil
```

Comment:
SetGStructOffset needs a CurrentThread, to have a current thread we need to have called updateThreadList, which calls addThread, which calls SwitchThread, which calls GetG, which needs GStructOffset to be set.